pkg/apis/servicecatalog/types.go

```diff
@@ -443,3 +457,19 @@ const (
 const (
 	FinalizerServiceCatalog string = "kubernetes-incubator/service-catalog"
 )
+
+// ParametersFromSource represents the source of a set of Parameters
+type ParametersFromSource struct {
```

Comment:
I think you got it wrong. The API looks like this:
```yaml
  ...
  parametersFrom:
    - secretKeyRef:
        name: mysecret
        key: mykey
```
where `parametersFrom` is the list of objects of type `ParameteresFromSource`:
```go
	// List of sources to populate parameters.
	// +optional
	ParametersFrom []ParametersFromSource
```